### PR TITLE
Allow use of a postfix3-compatible master.cf

### PR DESCRIPTION
**Pull request by jamesmollison**
Postfix3 (as distributed by Ubuntu xenial) chroots some daemons by default, and generates a large number of warnings if chroot=y is not set in master.cf

This PR gives users the option of shipping a postfix3-compatible master.cf where postfix3 is distributed by the operating system (so functionality is limited to xenial right now).  It should not change the default behaviour/output of the module. 

#167 is a dependancy of this PR. This allow params.pp to be included by (rather than inherited from) init.pp, so that $chroot_daemons can be set in the main postfix class.

Postfix3 will be included in the future release of most new operating systems, so I'm not sure how you want to deal with this going forward, this PR is to give compatibility to the current LTS release of Ubuntu.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -102,9 +105,9 @@
   $use_sympa           = false,         # postfix_use_sympa
   $postfix_ensure      = 'present',
   $mailx_ensure        = 'present',
-) inherits postfix::params {
-
+) { include ::postfix::params
```
Comment (raphink) [inline manifests/init.pp]: While I agree inheritance is not required here, I don't think it's in the scope of this PR to change it.

File: manifests/params.pp [hunk 2]
```diff
@@ -35,7 +36,11 @@
         default            => 'bsd-mailx',
       }
 
-      $master_os_template = "${module_name}/master.cf.debian.erb"
+      if $::lsbdistcodename == 'xenial' and $chroot_daemons {
```
Comment (raphink) [inline manifests/params.pp]: Should it fail when we're not on Xenial then?

Comment (jamesmollison) [inline manifests/params.pp]: The Boolean is false unless both conditions are met, which means no change in behaviour from present if the variable is not set (ie - the default master.cf.debian.erb is shipped).

Future debian & RHEL releases will have postfix3, so you may need to update this logic when they are released.

File: manifests/init.pp (hunk 1 above)
Comment (jamesmollison) [inline manifests/init.pp]: Have created #167 to cover this.  It will need to be approved first, this change will fail without it.

**Comment by jamesmollison** [comment]
Hi All - Any further issues with this PR?

File: manifests/init.pp (hunk 1 above)
Comment (raphink) [inline manifests/init.pp]: Why not?

Comment (jamesmollison) [inline manifests/init.pp]: If params (where the template is set) is inherited we can't pass the chroot_daemons variable to it, as params is evaluated first due to the inheritance.

Comment (raphink) [inline manifests/init.pp]: Then the standard is to expose `$chroot_daemons` in the main class and make it default to the value from params, and then use the variable from the main class. Cf https://docs.puppet.com/puppet/4.9/lang_classes.html#appendix-smart-parameter-defaults

Comment (jamesmollison) [inline manifests/init.pp]: Ah right. I didn't feel it was my place to refactor the whole module to accomodate small change this but maybe there isn't really any other option.  How do you want (me) to proceed here?

Comment (raphink) [inline manifests/init.pp]: OK then. I'll do that change and you can rebase your PR on it.

Comment (raphink) [inline manifests/init.pp]: The first step will be to merge #169 and make use of params.pp in there.

**Comment by jamesmollison** [comment]
This PR is better accomplished via #170 - closing this.

**Pull request closed without merging**